tests/functional/testplan/testing/test_listing.py

```diff
@@ -191,39 +230,53 @@ def test_programmatic_listing(
 
 
 @pytest.mark.parametrize(
-    "cmdline_args,expected_output",
+    "cmdline_args,prim_parts,expected_output",
     [
-        (["--list"], DEFAULT_NAME_OUTPUT),
-        (["--info", "pattern"], DEFAULT_PATTERN_OUTPUT),
-        (["--info", "name"], DEFAULT_NAME_OUTPUT),
+        (["--list"], 1, DEFAULT_NAME_OUTPUT),
+        (["--info", "pattern"], 1, DEFAULT_PATTERN_OUTPUT),
+        (["--info", "pattern"], 2, PATTERN_OUTPUT_WITH_PARTS),
+        (["--info", "name"], 1, DEFAULT_NAME_OUTPUT),
         (
             ["--info", "name", "--patterns", "*:Alpha", "*:Beta:test_c"],
+            2,
             to_stdout(
-                "Primary",
+                "Primary - part(0/2)",
                 "  Beta",
                 "    test_c",
                 "  Alpha",
                 "    test_c",
-                "    test_b",
                 "    test_a",
+                "Primary - part(1/2)",
+                "  Alpha",
+                "    test_b",
             ),
         ),
     ],
 )
-def test_command_line_listing(runpath, cmdline_args, expected_output):
-    multitest_x = MultiTest(name="Primary", suites=[Beta(), Alpha()])
-    multitest_y = MultiTest(name="Secondary", suites=[Gamma()])
+def test_command_line_listing(
+    runpath, cmdline_args, prim_parts, expected_output
+):
 
     with argv_overridden(*cmdline_args):
         plan = TestplanMock(name="plan", parse_cmdline=True, runpath=runpath)
 
         with captured_logging(TESTPLAN_LOGGER) as log_capture:
-            plan.add(multitest_x)
-            plan.add(multitest_y)
-
-            result = plan.run()
+            if prim_parts == 1:
+                plan.add(MultiTest(name="Primary", suites=[Beta(), Alpha()]))
+            else:
+                for i in range(prim_parts):
+                    plan.add(
+                        MultiTest(
+                            name="Primary",
+                            suites=[Beta(), Alpha()],
+                            part=(i, prim_parts),
+                        )
+                    )
+            plan.add(MultiTest(name="Secondary", suites=[Gamma()]))
```

Comment:
Could this be factored out to a factory function, which will be used to create the multitests for both testcase?